www/js/splash/pushNotifySettings.ts

```diff
@@ -0,0 +1,241 @@
+ *
+ * Other modules, such as the survey code, and the remotenotify module, listen
+ * to these CLOUD_NOTIFICATION_EVENTs and handle them through launching
+ * surveys, displaying popups, etc.
+ *
+ * This allows us to decouple the push handling logic from push notification
+ * interface. Note that the local notification is not currently decoupled since
+ * it only supports redirection to a specific app page. If the local
+ * notification handling gets more complex, we should consider decoupling it as well.
+ */
+
+import { updateUser } from '../commHelper';
+import { logDebug, displayError } from '../plugin/logger';
+import { publish, subscribe, EVENT_NAMES } from '../customEventHandler';
+import { isConsented, readConsentState } from './startprefs';
+import { readIntroDone } from '../onboarding/onboardingHelper';
+
+let push = null;
+
+/**
+ * @function initializes the PushNotification in window,
+ * assigns on 'notification' functionality
+ */
+const startupInit = function () {
+  push = window['PushNotification'].init({
+    ios: {
+      badge: true,
+      sound: true,
+      vibration: true,
+      clearBadge: true,
+    },
+    android: {
+      iconColor: '#008acf',
+      icon: 'ic_mood_question',
+      clearNotifications: true,
+    },
+  });
+  push.on('notification', function (data) {
+    if (window['cordova'].platformId == 'ios') {
+      // Parse the iOS values that are returned as strings
+      if (data && data.additionalData) {
+        if (data.additionalData.payload) {
+          data.additionalData.payload = JSON.parse(data.additionalData.payload);
+        }
+        if (data.additionalData.data && typeof data.additionalData.data == 'string') {
+          data.additionalData.data = JSON.parse(data.additionalData.data);
+        } else {
+          console.log('additionalData is already an object, no need to parse it');
+        }
+      } else {
+        logDebug('No additional data defined, nothing to parse');
+      }
+    }
+    publish(EVENT_NAMES.CLOUD_NOTIFICATION_EVENT, data);
+  });
+};
+
+/**
+ * @function registers notifications and handles result
+ * @returns Promise for initialization logic,
+ * resolves on registration with token
+ * rejects on error with error
+ */
+const registerPromise = function () {
+  return new Promise(function (resolve, reject) {
+    startupInit();
+    push.on('registration', function (data) {
+      console.log('Got registration ' + data);
+      resolve({
+        token: data.registrationId,
+        type: data.registrationType,
+      });
+    });
+    push.on('error', function (error) {
+      console.log('Got push error ' + error);
+      reject(error);
+    });
+    console.log('push notify = ' + push);
+  });
+};
+
+/**
+ * @function registers for notifications and updates user
+ * currently called on reconsent and on intro done
+ */
+const registerPush = function () {
+  registerPromise()
+    .then(function (t) {
+      // alert("Token = "+JSON.stringify(t));
```

Comment:
remove?

Reply:
That was there when I migrated over, I looked into the blame and didn't see any notes about it (and the change is 7 years old), so I think yes, safe to remove.

Reply:
done in [ad68e1b](https://github.com/e-mission/e-mission-phone/pull/1093/commits/ad68e1bb0e3a3387850ccc45912a3ce468184a37)